modin/backends/pandas/query_compiler.py

```diff
@@ -2603,24 +2610,50 @@ def _groupby_dict_reduce(
             query_compiler=self,
             by=by,
             axis=axis,
-            groupby_args=groupby_kwargs,
-            map_args=agg_kwargs,
-            reduce_args=agg_kwargs,
-            numeric_only=False,
+            groupby_kwargs=groupby_kwargs,
+            agg_args=agg_args,
+            agg_kwargs=agg_kwargs,
+            drop=drop,
+        )
+
+    def groupby_dtypes(
+        self,
+        by,
+        axis=0,
+        groupby_kwargs=None,
+        agg_args=None,
+        agg_kwargs=None,
+        drop=False,
+    ):
+        agg_func = (
+            (lambda grp: grp.dtypes)
+            if value_or_default(groupby_kwargs, dict).get("as_index", True)
+            else (lambda grp: grp.apply(lambda df: df.dtypes))
+        )
```

Comment:
this is a copy of the logic previously presented at the front-end:
https://github.com/modin-project/modin/blob/57bcfc130e9fbe93d41aa6403b80782ef49eb9db/modin/pandas/groupby.py#L286-L289
tests are failing without this condition